builder/azure/arm/config.go

```diff
@@ -1215,6 +1225,12 @@ func assertRequiredParametersSet(c *Config, errs *packersdk.MultiError) {
 		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("The os_type %q is invalid", c.OSType))
 	}
 
+	/////////////////////////////////////////////
+	// Storage
+	if c.Spot.EvictionPolicy != "" && c.Spot.EvictionPolicy != compute.VirtualMachineEvictionPolicyTypesDelete && c.Spot.EvictionPolicy != compute.VirtualMachineEvictionPolicyTypesDeallocate {
+		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("The spot.eviction_policy %q is not either empty, %q, or %q", c.Spot.EvictionPolicy, compute.VirtualMachineEvictionPolicyTypesDelete, compute.VirtualMachineEvictionPolicyTypesDeallocate))
+	}
+
```

Comment:
I think we should capture the edge case where max price is set and not an eviction policy, this is currently not blocked and could be a bit confusing if the user didn't understand Spot since they could set a max_price and it would do nothing, what do you think?
```suggestion
	if c.Spot.EvictionPolicy != "" {
		if c.Spot.EvictionPolicy != compute.VirtualMachineEvictionPolicyTypesDelete && c.Spot.EvictionPolicy != compute.VirtualMachineEvictionPolicyTypesDeallocate {
			s = packersdk.MultiErrorAppend(errs, fmt.Errorf("The spot.eviction_policy %q is invalid, eviction_policy must be %q, %q, or unset", c.Spot.EvictionPolicy, compute.VirtualMachineEvictionPolicyTypesDelete, compute.VirtualMachineEvictionPolicyTypesDeallocate))
		}
	} else {
		if c.Spot.MaxPrice != 0 {
			s = packersdk.MultiErrorAppend(errs, fmt.Errorf("Setting a spot.max_price without an spot.eviction_policy is invalid, eviction_policy must be %q or %q if max_price is set", compute.VirtualMachineEvictionPolicyTypesDelete, compute.VirtualMachineEvictionPolicyTypesDeallocate))
		}
	}

```

Reply:
good idea 👍 thanks. Added + added a test for this case.